apps/els_lsp/src/els_bsp_provider.erl

```diff
@@ -0,0 +1,262 @@
+-module(els_bsp_provider).
+
+-behaviour(els_provider).
+
+%% API
+-export([ start/1
+        , maybe_start/1
+        , info/1
+        , request/2
+        ]).
+
+%% els_provider functions
+-export([ is_enabled/0
+        , init/0
+        , handle_request/2
+        , handle_info/2
+        ]).
+
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include("els_lsp.hrl").
+-include_lib("kernel/include/logger.hrl").
+
+%%==============================================================================
+%% Types
+%%==============================================================================
+-type state() :: #{ running          := boolean()         % BSP server running?
```

Comment:
If I remember correctly, providers are registered by name. So, it may be simpler to check with a `erlang:is_process_alive(whereis(els_bsp_provider))` if the BSP provider is running or not. This would avoid the need of the `running` parameter and the `info` method.

Reply:
The BSP provider does run though even if the BSP server hasn't been started.